qtpy/tests/test_qtgui.py

```diff
@@ -101,8 +101,47 @@ def mouseDoubleClickEvent(self, event: QtGui.QMouseEvent) -> None:
     window.setMinimumSize(320, 240)  # ensure the window is of sufficient size
     window.show()
 
-    qtbot.mouseMove(window, QtCore.QPoint(42, 6 * 9))
-    qtbot.mouseDClick(window, QtCore.Qt.LeftButton)
+    with qtbot.waitExposed(window):
+        qtbot.mouseMove(window, QtCore.QPoint(42, 6 * 9))
+        qtbot.mouseDClick(window, QtCore.Qt.LeftButton)
+
+    # the rest of the functions are not actually tested
+    assert QtGui.QNativeGestureEvent.pos is not None
+    assert QtGui.QNativeGestureEvent.x is not None
+    assert QtGui.QNativeGestureEvent.y is not None
+    assert QtGui.QNativeGestureEvent.globalPos is not None
+    assert QtGui.QNativeGestureEvent.globalX is not None
+    assert QtGui.QNativeGestureEvent.globalY is not None
+    assert QtGui.QNativeGestureEvent.position is not None
+    assert QtGui.QNativeGestureEvent.globalPosition is not None
+    assert QtGui.QEnterEvent.pos is not None
+    assert QtGui.QEnterEvent.x is not None
+    assert QtGui.QEnterEvent.y is not None
+    assert QtGui.QEnterEvent.globalPos is not None
+    assert QtGui.QEnterEvent.globalX is not None
+    assert QtGui.QEnterEvent.globalY is not None
+    assert QtGui.QEnterEvent.position is not None
+    assert QtGui.QEnterEvent.globalPosition is not None
+    assert QtGui.QTabletEvent.pos is not None
+    assert QtGui.QTabletEvent.x is not None
+    assert QtGui.QTabletEvent.y is not None
+    assert QtGui.QTabletEvent.globalPos is not None
+    assert QtGui.QTabletEvent.globalX is not None
+    assert QtGui.QTabletEvent.globalY is not None
+    assert QtGui.QTabletEvent.position is not None
+    assert QtGui.QTabletEvent.globalPosition is not None
+    assert QtGui.QHoverEvent.pos is not None
+    assert QtGui.QHoverEvent.x is not None
+    assert QtGui.QHoverEvent.y is not None
+    assert QtGui.QHoverEvent.position is not None
+    assert QtGui.QTabletEvent.pos is not None
+    assert QtGui.QTabletEvent.x is not None
+    assert QtGui.QTabletEvent.y is not None
+    assert QtGui.QTabletEvent.globalPos is not None
+    assert QtGui.QTabletEvent.globalX is not None
+    assert QtGui.QTabletEvent.globalY is not None
+    assert QtGui.QTabletEvent.position is not None
+    assert QtGui.QTabletEvent.globalPosition is not None
```

Comment:
This is getting a bit code smelly. Could you maybe replace this with a two-layer nested `for` loop using `getattr`?

```suggestion
    for name in ['QNativeGestureEvent', 'QEnterEvent', 'QTabletEvent', 'QHoverEvent']:
        for attr in ['pos', 'x', 'y', 'globalPos', 'globalX', 'globalY', 'position', 'globalPosition']:
            assert getattr(QtGui, name) is not None
```

Also, `QTabletEvent` is repeated twice. Did you mean to include something else, like `QSinglePointEvent`?

Reply:
Oh, the lines are just an extra copy. Initially, I thought of adding `QWheelEvent` to the list. As for your suggestion, it's not equivalent to my code, although close. I'll fix the code in and hour.